sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -244,7 +244,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def json(path: String): DataFrame = {
     // This method ensures that calls that explicit need single argument works, see SPARK-16009
-    json(Seq(path): _*)
+    option("path", path).json(Seq.empty: _*)
```

Comment:
@zsxwing I see your point. The issue is from the others who build the data sources using the data source APIs. See the discussion in https://github.com/apache/spark/pull/13727#discussion_r68624316